cargo-public-api/src/main.rs

```diff
@@ -95,12 +103,34 @@ pub struct Args {
 fn main_() -> Result<()> {
     let args = get_args();
 
-    if let Some(commits) = &args.diff_git_checkouts {
-        print_public_items_diff_between_two_commits(&args, commits)
+    let diff = if let Some(commits) = &args.diff_git_checkouts {
+        Some(print_diff_between_two_commits(&args, commits)?)
     } else if let Some(files) = &args.diff_rustdoc_json {
-        print_public_items_diff_between_two_rustdoc_json_files(&args, files)
+        Some(print_diff_between_two_rustdoc_json_files(&args, files)?)
     } else {
-        print_public_items_of_current_commit(&args)
+        print_public_items_of_current_commit(&args)?;
+        None
+    };
+
+    check_diff(&args, diff)
+}
+
+fn check_diff(args: &Args, diff: Option<PublicItemsDiff>) -> Result<()> {
+    match (&args.deny, diff) {
+        // We were requested to deny diffs, so make sure there is no diff
+        (Some(_deny), Some(diff)) => {
+            if diff.is_empty() {
+                Ok(())
+            } else {
+                Err(anyhow!(error::Error::DiffDenied))
+            }
```

Comment:
```suggestion
            diff.is_empty().then(|| Ok(())).unwrap_or_else(|| anyhow!(error::Error::DiffDenied))
```

What about (modulo some linebreaks maybe) :laughing:

Reply:
Well, apart from that it is more complicated (`bool.then()` returns an `Option<T>`, but there is no need for that here I think) and does not compile, we could use that instead I guess 😅 

Why not use the current code though? Don't you think it is easier to see what happens? 

If you strongly prefer it to be something else I will for sure consider it, but I'd have to see something that compiles first :) (My deepest apologies if it does compile and I just misunderstood how to apply the change)

But I have to be honest: I strongly prefer simplicity over "smartness" (in lack of a better word).

Reply:
> Why not use the current code though? Don't you think it is easier to see what happens?

I just like to explore all options in terms of "idomadicness" :smile: 

I'm okay with the current implementation of course, I consider this resolved.